lib/src/rules/tighten_type_of_initializing_formals.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Tighten type of initializing formal.';
+
+const _details = r'''
+
+Tighten type of initializing formal if an non null assert exists.
+
```

Comment:
A little more rationale would be great here if possible.  Why are we recommending this?

Reply:
Is the following better:

> Tighten type of initializing formal if a non-null assert exists. Using a
non-nullable type will provide static warnings in the analyzer instead of
runtime errors.

Reply:
To allow the type system to catch problems rather than have them only be caught at run-time. See #2237 for more discussion.

Reply:
Updated with Brian comment :)

Reply:
Thanks!